src/OrleansProviders/Streams/Common/PooledCache/FixedSizeObjectPool.cs

```diff
@@ -52,16 +52,25 @@ public override T Allocate()
             T obj;
             lock (locker)
             {
-                // if we've used all we are allowed, signal object it has been purged and should be returned to the pool
-                if (usedObjects.Count >= maxObjectCount)
+                // just send signal to EventhubQueueCaches to ask them do a TimeBasedPurge
```

Comment:
this is generally usable, just Eventhub cache will handle this in its own manner. and other cache will handle it as before. But I agree the comments can be better. But I kind want comment here to explain to how EventHUb will handle this in its own manner to reduce confusion for other engineers to look at